docs/manual/src/udl/custom_types.md

```diff
@@ -1,72 +1,107 @@
 # Custom types
 
-Custom types allow you to extend the UniFFI type system to support types from your Rust crate or 3rd
-party libraries. This works by converting to and from some other UniFFI type to move data across the
-FFI. You must provide a `UniffiCustomTypeConverter` implementation to convert the types.
+Custom types offer a way to use user-defined types in your interface that don't derive one of the UniFFI type traits (`uniffi::Record`, `uniffi::Enum`, `uniffi::Object`, etc.).
+Instead, custom types are converted to/from a existing type that does derive those traits when being passed across the FFI.
+
+Custom types are often used with structs that wraps a primitive type, for example `Guid(String)`.
+These types can be passed across the FFI as the primitive type, which can be more efficient than passing a struct.
+The foreign bindings will treat these types as the primitive type, for example `Guid` could appear a string to the foreign code.
+
+Custom types can also be customized on the foreign side.  For example, a URL could be:
+* Represented by the `url::Url` type in Rust
+* Passed across the FFI as a string
+* Represented by the `java.net.URL` type in Kotlin
+
+Lastly, the `Guid(String)` example shows another benefit of custom types: they can be used to
+support Rust types that are not currently supported by UniFFI like tuple-style structs.
 
 ## Custom types in the scaffolding code
 
-Consider the following trivial Rust abstraction for a "handle" which wraps an integer:
+### custom_type!
+
+Use the `custom_type!` macro to define a new custom type.
 
 ```rust
-pub struct Handle(i64);
+
+// Some complex struct that can be serialized/deserialized to a string.
+use some_mod::SerializableStruct;
+
+/// `SerializableStruct` objects will be passed across the FFI the same way `String` values are.
+uniffi::custom_type!(SerializableStruct, String);
 ```
 
-In this trivial example, the simplest way to expose this is with a macro.
+By default:
+
+ - Values passed to the foreign code will be converted using `<SerializableStruct as Into<String>>` before being lowered as a `String`.
+ - Values passed to the Rust code will be converted using `<String as TryInto<SerializableStruct>>` after lifted as a `String`.
+ - The `TryInto::Error` type can be anything that implements `Into<anyhow::Error>`.
```

Comment:
My brain hurts - following on from the last comment - have we got these names correct? We have `into_existing` which doesn't return a result, but here talking about "TryInto" (which does) seems confusing and backwards.

I'm worried my last suggestion which gave us these new names actually made things more complicated?

Reply:
I think I just got this backwards and I'll reverse them.